### PR TITLE
Fix Blockchain.Persist

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by shargon** [comment]
which plugin gave the problem?

**Comment by erikzhang** [comment]
Plugins don't cause problems. It is that `SnapshotCache` cannot be reused after commit, and a new instance should be created. So subsequent read-only operations should be read from `NeoSystem.StoreView`, otherwise the data in the old snapshot may be read.

**Review by shargon — APPROVED** [review]

**Review by Liaojinghui — APPROVED** [review]

File: src/neo/Plugins/IPersistencePlugin.cs [hunk 1]
```diff
@@ -35,8 +35,7 @@ public interface IPersistencePlugin
         /// </summary>
         /// <param name="system">The <see cref="NeoSystem"/> object that contains the blockchain.</param>
         /// <param name="block">The block being persisted.</param>
-        /// <param name="snapshot">The snapshot used for persistence.</param>
-        void OnCommit(NeoSystem system, Block block, DataCache snapshot) { }
+        void OnCommit(NeoSystem system, Block block) { }
```
Comment (devhawk) [inline src/neo/Plugins/IPersistencePlugin.cs:38]: Won't think break existing persistence plugins that implement OnCommit?

Comment (devhawk) [reply src/neo/Plugins/IPersistencePlugin.cs:38]: this will be part of 3.3?

Comment (erikzhang) [reply src/neo/Plugins/IPersistencePlugin.cs:38]: Yes.

**Pull request merged**